nixos/modules/services/networking/teleport.nix

```
@@ -0,0 +1,88 @@
+{ config, pkgs, lib, ... }:
+
+with lib;
+
+let
+  cfg = config.services.teleport;
+  pidFilePath = "/run/teleport.pid";
+  configJSON = pkgs.writeText "teleport.json" (builtins.toJSON cfg.settings);
+  cmdlineArgs = [ "--pid-file ${pidFilePath}" ]
+    ++ optional cfg.debug.enable "--debug"
+    ++ optional cfg.diag.enable "--diag-addr=${cfg.diag.addr}:${toString cfg.diag.port}"
+    ++ optional (cfg.settings != { }) "--config=${configJSON}"
+    ++ cfg.extraOpts;
+in
+{
+  options = {
+    services.teleport = with lib.types; {
+      enable = mkEnableOption "Teleport service.";
+
+      settings = mkOption {
+        type = attrs;
+        default = { };
+        example = literalExpression ''
+          {
+            teleport = {
+              nodename = "client";
+              advertise_ip = "192.168.1.2";
+              auth_token = "60bdc117-8ff4-478d-95e4-9914597847eb";
+              auth_servers = [ "192.168.1.1:3025" ];
+          }
+        '';
+        description = "Reference: https://goteleport.com/docs/setup/reference/config/";
```

this could do with a more elaborate description. maybe something like this?
```suggestion
        description = literalDocBook ''
          Contents of the <literal>teleport.json</literal> config file.
          The <literal>--config</literal> arguments will only be passed if this set is not empty.
          
          Reference: https://goteleport.com/docs/setup/reference/config/
        '';
```

---

Don't forget to use the syntax that makes links clickable! Users will love you ❤️